File: ROS/ROB-115/Python/loops.py
import time

# Define an array
my_array = [10, 20, 30, 40, 50]

# loop
for i in my_array:
    print(i)

print('===============')

# range loop
for i in range(0,5):
    #print(i*10)
    print(my_array[i])

print('===============')

# while loop
index = 0
while True:
    if index > 4:
        break
    print(my_array[index])
    index+=1

print('===============')

# while loop
index = 0
while True:
    print(my_array[index]) if index < 5 else print('test')
    index+=1
    if index > 6:
        break

File: ROS/ROB-115/C++/loops.cpp
#include <iostream>
#include <vector>

using namespace std;

// Iterates through the vector using a range-based for loop.
// Preferred when index access is not required for clarity and readability.
void range_based_loop(const vector<int>& my_array) 
{
    for (auto i : my_array)  
        cout << i << " ";  
    cout << endl;
}

// Iterates through the vector using an indexed for loop.
// Useful when index-based operations or modifications are needed.
void standard_loop(const vector<int>& my_array) 
{
    for (size_t i = 0; i < my_array.size(); i++)  
        cout << my_array[i] << " ";  
    cout << endl;
}

int main()
{
    vector<int> my_array = {1, 2, 3, 4, 5, 6, 6, 4, 6};

    range_based_loop(my_array);

    cout << "==============" << endl;

    standard_loop(my_array);

    return 0;
}
